fix(vim-patch.sh): don't treat 1-file runtime-only patch as N/A

zeertzjq opened the pull request:
If the patch in question does not change src/version.c, `git log -1`
instead shows the previous patch that changes src/version.c, which
causes 1-file runtime-only patches to be considered N/A.

Check that the remaining file is src/version.c to fix this problem.
Also use `git diff-tree` instead of `git log -1` for version.c.

zeertzjq committed:
fix(vim-patch.sh): don't treat 1-file runtime-only patch as N/A

If the patch in question does not change src/version.c, `git log -1`
instead shows the previous patch that changes src/version.c, which
causes 1-file runtime-only patches to be considered N/A.

Check that the remaining file is src/version.c to fix this problem.
Also use `git diff-tree` instead of `git log -1` for version.c.

zeertzjq commented:
cc @janlazo

janlazo commented:
That section assumes the only remaining unfiltered file is `version.c`. Switching to `git show` is to compensate for the insufficient check from `FILE_LNUM`.

I approve.

But, I'd rather correct the `$FILE_LNUM` check to also require that the remaining file is `version.c` to not need further workarounds.

zeertzjq commented:
It seems that `git diff-tree` also doesn't have this problem. I changed `git show` to `git diff-tree` like the check above.

I also modified the `FILE_LNUM` check to only enter the `if` branch when src/version.c is the only remaining file. I don't think there is any harm in using `git diff-tree` instead of `git log -1` though.

janlazo commented:
https://github.com/janlazo/neovim/commit/2b6fd97a6e98e75907463929f44e88f4c44b244c is what I had in mind.

janlazo commented on scripts/vim-patch.sh:
```diff
@@ -892,21 +892,20 @@ is_na_patch() {
-  if test "$FILE_LNUM" -eq 1; then
+  local FILES_REMAINING
+  FILES_REMAINING="$(diff <(git -C "${VIM_SOURCE_DIR}" diff-tree --no-commit-id --name-only -r "$patch" | grep -v -f "$NA_REGEXP") "$NA_FILELIST" |
+    grep '^<')"
```
`||` is to handle empty string for this `grep` or broken pipe due to `diff` getting an empty file. Had issues with is_na_patch() aborting early without `||`.

zeertzjq replied:
I can't seem to reproduce the problem, but I'll add a `|| true`.
